core/metric.go

```diff
@@ -64,6 +67,31 @@ func (n Namespace) Strings() []string {
 	return ns
 }
 
+// getSeparator returns an valid separator from nsPriorityList.
+// Otherwise the core separator is returned.
+func (n Namespace) getSeparator() string {
+	mp := n.toMap()
```

Comment:
The idea of using a map for through characters is great.

Number of characters from namespaces are variable, and characters change from a namespace to another. What if instead of generating this map we use a static map of the priority characters?

This is globaly the same algorithm, here is an implementation I quickly wrote in a `test.go` file:
```
package main

import (
	"fmt"

	"github.com/intelsdi-x/snap/core"
)

var (
	nsPriorityList = map[string]bool{"/": false, "-": false, "_": false, "㊽": false, "Ä": false, "大": false, "小": false, "ᵹ": false, "☍": false, "ヒ": false}
)

type Namespace core.Namespace

func main() {
	//ns := core.NewNamespace("p大Älo-p", "ヒwo㊽rld/", "te☍_st小ᵹ")
	//ns := core.NewNamespace("p大Älo-p", "wo㊽rld/", "te_st")
	ns := core.NewNamespace("plop", "world", "test")

	fmt.Println(ns.GetSeparator())
}

// getSeparator returns an valid separator from nsPriorityList.
// Otherwise the core separator is returned.
func (n Namespace) GetSeparator() string {
	for _, elem := range n {
		for _, r := range elem.Value {
			s := fmt.Sprintf("%c", r)
			if _, ok := nsPriorityList[s]; ok && !nsPriorityList[s] {
				nsPriorityList[s] = true
			}
		}
	}
	for sep, used := range nsPriorityList {
		if !used {
			return sep
		}
	}
	return core.Separator
}
```

`go run test.go`